kolibri/plugins/coach/assets/src/app.js

```diff
@@ -83,6 +83,18 @@ class CoachToolsModule extends KolibriApp {
       } else {
         next();
       }
+
+      if (!this.store.getters.isUserLoggedIn) {
```

Comment:
Hrm, it might be worth investigating the `withAuthMessage` component defined here: https://github.com/learningequality/kolibri/blob/4dae6286e2137cfad17326870f9706ce5982d18a/kolibri/plugins/device/assets/src/views/withAuthMessage.js#L17 (search the code base for other uses), alternatively, the AuthMessage component can be used inside the Coach index page.

This particular implementation will kind of work, although I am fairly sure the `next` query parameter will be lost during the redirect to the auth page.

It is also possible for Kolibri to be configured to redirect unauthenticated users to the Learn page, so in that case, this redirect will not prompt someone to login.

Reply:
@GarvitSinghal47 -- would you be able to revise this PR according to the feedback above?

Reply:
@nucleogenesis , my apologies for the oversight. I completely forgot about this open pull request. I will make the necessary adjustments and provide the updates promptly.